csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -170,6 +170,132 @@ public void Format(IMessage message, TextWriter writer)
         }
 
         /// <summary>
+        /// Formats the specified field of the message as JSON.
+        /// </summary>
+        /// <param name="message">The message to format.</param>
+        /// <param name="field">The field in the message to format.</param>
+        /// <returns>The formatted field.</returns>
+        public string FormatFieldValue(IMessage message, FieldDescriptor field)
+        {
+            var writer = new StringWriter();
+            FormatFieldValue(message, field, writer);
+            return writer.ToString();
+        }
+
+        /// <summary>
+        /// Formats the specified field of the message as JSON.
+        /// </summary>
+        /// <param name="message">The message to format.</param>
+        /// <param name="field">The field in the message to format.</param>
+        /// <param name="index">The index of the repeated message to access.</param>
+        /// <returns>The formatted field.</returns>
+        public string FormatRepeatedFieldValue(IMessage message, FieldDescriptor field, int index)
```

Comment:
This only accepts repeated (non map) fields. The index is used to select a field before formatting.